packages/server/routes/api/guilds.js

```diff
@@ -26,29 +27,179 @@ router.get(
+      if (error.name === "PrismaClientKnownRequestError") {
+        switch (error.code) {
+          case "P2023":
+            return response.status(400).json({
+              status: "fail",
+              data: {
+                guildId: "Provided guild ID is not a valid UUID.",
+              },
+            });
+
+          case "P2025":
+            return response.status(404).json({
+              status: "fail",
+              data: {
+                guildId: `No guild with an ID of ${request.params.guildId} could be found.`,
+              },
+            });
+
+          default:
+            return response.status(500).json({
+              status: "fail",
+              errorName: error.name,
+              errorMessage: error.message,
+            });
+        }
+      } else {
+        return response.status(500).json({
+          status: "fail",
+          errorName: error.name,
+          errorMessage: error.message,
+        });
+      }
+    }
+  }
+);
+
+// Create Guild
+router.post("/", AuthController.authenticate, async (request, response) => {
+  try {
+    return response.status(200).json({
+      status: "success",
+      data: {
+        createdGuild: await GuildController.createGuild(request.body),
+      },
+    });
+  } catch (error) {
+    // Note: When errored, this route returns a huge error message, the end of the message contains missing arugments/fields required for the route to function properly.
+    return response.status(400).json({
+      status: "error",
+      errorName: error.name,
+      message: error.message,
+    });
+  }
+});
+
+// Update guild by ID
+router.put(
+  "/:guildId",
+  AuthController.authenticate,
+  async (request, response) => {
+    try {
+      const { guildId } = request.params;
+      const guildData = request.body;
+
+      return response.status(200).json({
         status: "success",
         data: {
-          guild,
+          updatedGuild: await GuildController.updateGuild(guildId, guildData),
         },
       });
     } catch (error) {
-      response.status(500).json({
-        status: "error",
-        message: error.message,
+      if (error.name === "PrismaClientKnownRequestError") {
+        switch (error.code) {
+          case "P2023":
+            return response.status(400).json({
+              status: "fail",
+              data: {
+                guildId: "Provided guild ID is not a valid UUID.",
+              },
+            });
+
+          case "P2025":
+            return response.status(404).json({
+              status: "fail",
+              data: {
+                guildId: `No guild with an ID of ${request.params.guildId} could be found.`,
+              },
+            });
+
+          default:
+            return response.status(500).json({
+              status: "fail",
+              errorName: error.name,
+              errorMessage: error.message,
+            });
+        }
+      } else {
+        return response.status(500).json({
```

Comment:
Guild updating also needs similar error handling like I said for the guild creation route. Check for `error.name` of `PrismaClientValidationError` and then send an error message back like "Guild could not be updated because of missing or invalid arguments`.